python/ray/worker.py

```diff
@@ -677,7 +691,13 @@ def _get_arguments_for_execution(self, function_name, serialized_args):
                     # error message here.
                     raise RayGetArgumentError(function_name, i, arg, argument)
                 elif isinstance(argument, ray.actor.ActorHandleWrapper):
-                    argument = ray.actor.unwrap_actor_handle(self, argument)
+                    argument = argument.unwrap(self)
+                    # This task was passed an actor handle as an argument. Add
+                    # it to the registry of active handles passed to this task.
+                    # The entry will be deleted when the actor handle goes out
+                    # of scope.
+                    self.passed_actor_handles[
+                        argument._ray_actor_id] = argument
```

Comment:
Why is the handling here so much simpler than in `ray.get`? Since they both call `get_object` under the hood I would have expected them to do the same thing.

Reply:
Hmm yeah, actually I'll move the code currently in `ray.get` to `get_object`. I think what I have is incorrect at the moment since the return values won't get unwrapped if they are passed as an argument to another remote function.

I originally kept them separate because I figured the part where we join handles together is specific to `ray.get`.